tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -32,254 +32,198 @@ var _ = Describe("odo devfile push command tests", func() {
 		helper.CommonAfterEach(commonVar)
 	})
 
-	Context("verify devfile push works", func() {
-
-		It("checks that odo push works with a devfile", func() {
+	When("creating a nodejs compoenet", func() {
+		output := ""
+		BeforeEach(func() {
 			helper.Cmd("odo", "create", "nodejs", "--project", commonVar.Project, cmpName).ShouldPass()
-
 			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
-			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
-
-			helper.Cmd("git", "init").ShouldPass()
-			remote := "origin"
+		})
+		When("setting git conifg and running odo push", func() {
 			remoteURL := "https://github.com/odo-devfiles/nodejs-ex"
-			helper.Cmd("git", "remote", "add", remote, remoteURL).ShouldPass()
+			BeforeEach(func() {
+				helper.Cmd("git", "init").ShouldPass()
+				remote := "origin"
+				helper.Cmd("git", "remote", "add", remote, remoteURL).ShouldPass()
+				helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
+				output = helper.Cmd("odo", "push", "--project", commonVar.Project).ShouldPass().Out()
+			})
 
-			output := helper.Cmd("odo", "push", "--project", commonVar.Project).ShouldPass().Out()
-			Expect(output).To(ContainSubstring("Changes successfully pushed to component"))
+			It("checks that odo push works with a devfile", func() {
+				Expect(output).To(ContainSubstring("Changes successfully pushed to component"))
+			})
+
+			It("check annotations from the deployment after odo push", func() {
 
-			annotations := commonVar.CliRunner.GetAnnotationsDeployment(cmpName, "app", commonVar.Project)
-			var valueFound bool
-			for key, value := range annotations {
-				if key == "app.openshift.io/vcs-uri" && value == remoteURL {
-					valueFound = true
+				annotations := commonVar.CliRunner.GetAnnotationsDeployment(cmpName, "app", commonVar.Project)
+				var valueFound bool
+				for key, value := range annotations {
+					if key == "app.openshift.io/vcs-uri" && value == remoteURL {
+						valueFound = true
+					}
 				}
-			}
-			Expect(valueFound).To(BeTrue())
+				Expect(valueFound).To(BeTrue())
+			})
 
-			// update devfile and push again
-			helper.ReplaceString("devfile.yaml", "name: FOO", "name: BAR")
-			helper.Cmd("odo", "push", "--project", commonVar.Project).ShouldPass()
-		})
+			When("updating a variable into devfile", func() {
```

Comment:
Why do we need this extra layer of `when` . I think there is only one scenario uses `BeforeEach` steps. I would suggest using and `It` spec instead of `when`